src/main/java/no/nav/fo/veilarbregistrering/arbeidsforhold/adapter/ProxyArbeidsforholdGateway.java

```diff
@@ -0,0 +1,67 @@
+package no.nav.fo.veilarbregistrering.arbeidsforhold.adapter;
+
+import no.nav.fo.veilarbregistrering.arbeidsforhold.ArbeidsforholdGateway;
+import no.nav.fo.veilarbregistrering.arbeidsforhold.FlereArbeidsforhold;
+import no.nav.fo.veilarbregistrering.bruker.Foedselsnummer;
+import no.nav.sbl.featuretoggle.unleash.UnleashService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProxyArbeidsforholdGateway implements ArbeidsforholdGateway {
+
+    private final static Logger LOG = LoggerFactory.getLogger(ProxyArbeidsforholdGateway.class);
+
+    private final ArbeidsforholdGateway soapArbeidsforholdGateway;
+    private final ArbeidsforholdGateway restArbeidsforholdGateway;
+    private final UnleashService unleashService;
+
+    public ProxyArbeidsforholdGateway(
+            ArbeidsforholdGateway soapArbeidsforholdGateway,
+            ArbeidsforholdGateway restArbeidsforholdGateway,
+            UnleashService unleashService) {
+        this.soapArbeidsforholdGateway = soapArbeidsforholdGateway;
+        this.restArbeidsforholdGateway = restArbeidsforholdGateway;
+        this.unleashService = unleashService;
+    }
+
+    @Override
+    public FlereArbeidsforhold hentArbeidsforhold(Foedselsnummer fnr) {
+        FlereArbeidsforhold arbeidsforholdFraSoap = hentArbeidsforholdFraSoap(fnr);
+
+        if (arbeidsforholdRestIsEnabled()) {
+            FlereArbeidsforhold arbeidsforholdFraRest = hentArbeidsforholdFraRest(fnr);
+
+            if (arbeidsforholdFraSoap.equals(arbeidsforholdFraRest)) {
```

Comment:
Kult pattern dettte

Reply:
Altid en fare for at det blir mye greier for lite, men liker at jeg kan teste det uten å endre kode i eksisterende kode, og når ting er verifisert, kan vi slette proxy og gammel gateway. Da står vi igjen med bare en implementasjon, og da vil spring finne riktig versjon ut av boksen.